src/backend/utils/misc/queryenvironment.c

```diff
@@ -151,6 +153,9 @@ void remove_queryEnv() {
 	list_free(currentQueryEnv->dropped_namedRelList);
 	currentQueryEnv->dropped_namedRelList = NIL;
 
+	list_free(currentQueryEnv->savedCatcacheMessages);
```

Comment:
let us put all babelfish related cleanup in a hook

Reply:
re this and above comment about separating ENR babelfish and PG logic - let's do this in a separate PR. i think it would bloat this change too much. there are a few things here we can clean up at once, like moving many of the BBF specific functions to hooks, cleaning up the hook file for all of the temp table functions, normalizing naming, and as you mention, making the queryEnv initialization and cleanup into BBF specific hooks.